bv-evaluation/Snakefile

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import random
+import subprocess
+import concurrent.futures
+import shutil
+import multiprocessing
+import psutil
+import time
+import threading
+import platform
+from functools import partial
+from pathlib import Path
+import sys
+from runwithlimits import *
+from snakelib import *
+
+configfile: "config.yaml"
+workdir: "../"
+
+gitroot=get_git_root()
+sed=get_sed()
+HACKERSDELIGHT_FILE_NAMES, = glob_wildcards(gitroot / "SSA/Projects/InstCombine/HackersDelight/{file}.lean")
+hdel_nreps = config["hdel_nreps"]
+
+onstart:
+    shell("elan --version")
+    shell("cd {gitroot} && lake exe cache get && lake build")
+    shell("uv --version")
+    shell("bitwuzla --version")
```

Comment:
Should this be redirecting outputs to log files?

You might also want to print some prelude information before dumping `elan --version` straight to the user.